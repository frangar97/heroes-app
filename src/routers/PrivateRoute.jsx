import { AuthContext } from 'auth/authContext'
import React from 'react'
import { useContext } from 'react'
import { Navigate } from 'react-router';

export const PrivateRoute = (props) => {
    const { user } = useContext(AuthContext);

    return user.logged ? props.children : <Navigate to="/login" />
}
