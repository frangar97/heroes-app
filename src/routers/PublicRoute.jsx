import { AuthContext } from 'auth/authContext'
import React from 'react'
import { useContext } from 'react'
import { Navigate } from 'react-router';

export const PublicRoute = (props) => {
    const { user } = useContext(AuthContext);

    return user.logged ? <Navigate to="/marvel" /> : props.children;
}
