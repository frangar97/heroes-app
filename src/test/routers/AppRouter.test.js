import { AuthContext } from "auth/authContext";
import { mount } from "enzyme";
import { AppRouter } from "routers/AppRouter";

describe("Pruebas en el componente <AppRouter/>", () => {

    const contextValue = {
        user: {
            logged: false
        }
    }

    test("Debe mostrar el login si no esta autenticado", () => {
        const wrapper = mount(<AuthContext.Provider value={contextValue}><AppRouter /></AuthContext.Provider>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h1').text().trim()).toBe("Login");
    })

})