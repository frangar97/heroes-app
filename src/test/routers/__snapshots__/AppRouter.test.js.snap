// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AppRouter/> Debe mostrar el login si no esta autenticado 1`] = `
Array [
  " ",
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>,
]
`;
